### PR TITLE
Make ConnectionMode for CosmosDB editable by consumer

### LINKED ISSUE #16842: 3.0 Preview 7: ConnectionMode to Cosmos DB is hard-coded to Direct
The Entity Framework 3.0 Preview connects to Cosmos DB using the Direct connection mode and this is non configurable by the consumer of the API. While I understand that this is the recommended connection mode to Cosmos DB moving forward, we're using Istio as a service mesh and we allow connections outside the Kubernetes cluster on a whitelist basis. Because Istio doesn't allow to specify port ranges, this would force us to write a list of all the ports in the 10,000-20,000 to allow the API to connect to the database.

While this is not a bug, it would be nice if in the future the client was allowed to alter the default connection mode to Cosmos DB.

### Further technical details
EF Core version: 3.0.0-preview7.19362.6
Database Provider: Microsoft.EntityFrameworkCore.Cosmos
Operating system: Windows 10
IDE: Visual Studio 2019 16.1.6

### PR DESCRIPTION
**Pull request by chiasottis**
- Expose CosmosDB connection mode to the library consumer as a CosmosOptions propriety.

Fixes #16842

### CODE REVIEW

File: src/EFCore.Cosmos/Infrastructure/CosmosDbContextOptionsBuilder.cs [hunk 1]
```diff
@@ -50,6 +51,13 @@ public CosmosDbContextOptionsBuilder([NotNull] DbContextOptionsBuilder optionsBu
         public virtual CosmosDbContextOptionsBuilder Region([NotNull] string region)
             => WithOption(e => e.WithRegion(Check.NotNull(region, nameof(region))));
 
+        /// <summary>
+        /// Configures the context to use the provide connection mode.
```
Comment (AndriySvyryd) [inline src/EFCore.Cosmos/Infrastructure/CosmosDbContextOptionsBuilder.cs]:
```suggestion
        /// Configures the context to use the provided connection mode.
```

Comment (chiasottis) [reply src/EFCore.Cosmos/Infrastructure/CosmosDbContextOptionsBuilder.cs]: I fixed the typo.

**Comment by chiasottis** [comment]
@AndriySvyryd, I added the connection mode to the CosmosDbOptionExtension.ExtensionInfo.GetServiceProviderHashCode() method.

**Comment by AndriySvyryd** [comment]
Thanks for your contribution!

**Pull request merged**